nilmtk/api.py

```diff
@@ -241,60 +242,108 @@ def train_jointly(self,clf,d):
+	                    print ("Creating an Artificial Aggregate")
+	                    test_mains = pd.DataFrame(np.zeros(appliance_readings[0].shape),index = appliance_readings[0].index,columns=appliance_readings[0].columns)
+	                    for app_reading in appliance_readings:
+	                        test_mains+=app_reading
+
+
+	                self.test_mains = [test_mains]
+	                for i, appliance_name in enumerate(self.appliances):
+	                    self.test_submeters.append((appliance_name,[appliance_readings[i]]))
+	                
+	                self.storing_key = str(dataset) + "_" + str(building) 
+	                self.call_predict(self.classifiers)
+
+    # def sitetest_jointly(self,d):
+
+
+    #     # store the test_main readings for all buildings
+    #     for dataset in d:
+    #         print("Loading data for ",dataset, " dataset")
+    #         test=DataSet(d[dataset]['path'])
+    #         for building in d[dataset]['buildings']:
+    #             test.set_window(start=d[dataset]['buildings'][building]['start_time'],end=d[dataset]['buildings'][building]['end_time'])
+    #             test_mains=next(test.buildings[building].elec.mains().load(physical_quantity='power', ac_type='apparent', sample_period=self.sample_period))
+    #             # print(f'test_mains: {test_mains}')
 
+                
+    #             if self.DROP_ALL_NANS:
+    #                 test_mains= self.dropna(test_mains)
 
-                self.test_mains = [test_mains]
-                for i, appliance_name in enumerate(self.appliances):
-                    self.test_submeters.append((appliance_name,[appliance_readings[i]]))
                 
-                self.storing_key = str(dataset) + "_" + str(building) 
-                self.call_predict(self.classifiers)
+
+    #             self.test_mains = [test_mains]
+    #             self.storing_key = str(dataset) + "_" + str(building) 
+    #             df = self.call_predict(self.classifiers)
+    #     return df
             
 
-    def dropna(self,mains_df, appliance_dfs):
+    def dropna(self,mains_df, appliance_dfs=None):
         """
         Drops the missing values in the Mains reading and appliance readings and returns consistent data by copmuting the intersection
         """
         print ("Dropping missing values")
 
         # The below steps are for making sure that data is consistent by doing intersection across appliances
-        mains_df = mains_df.dropna()
-        for i in range(len(appliance_dfs)):
-            appliance_dfs[i] = appliance_dfs[i].dropna()
-        ix = mains_df.index
-        for  app_df in appliance_dfs:
-            ix = ix.intersection(app_df.index)
-        mains_df = mains_df.loc[ix]
-        new_appliances_list = []
-        for app_df in appliance_dfs:
-            new_appliances_list.append(app_df.loc[ix])
-        return mains_df,new_appliances_list
+        if appliance_dfs == None:
```

Comment:
See my comments above about duplicate code in `if` cases and using `[]` instead of `None`.